recipes/boost/bld.bat

```diff
@@ -1,14 +1,18 @@
 :: Set the right msvc version according to Python version
-if "%PY_VER%"=="2.7" (
-    set MSVC_VER=9.0
-    set LIB_VER=90
-) else if "%PY_VER%"=="3.4" (
-    set MSVC_VER=10.0
-    set LIB_VER=100
-) else (
-    set MSVC_VER=14.0
-    set LIB_VER=140
-)
+REM write a temporary batch file to map cl.exe version to visual studio version
+echo @echo 15=9 > msvc_versions.bat
+echo @echo 16=10 >> msvc_versions.bat
+echo @echo 17=11 >> msvc_versions.bat
+echo @echo 18=12 >> msvc_versions.bat
+echo @echo 19=14 >> msvc_versions.bat
```

Comment:
Could you please give a little explanation of what is happening here? Particularly for people who are less familiar with Windows like myself. :smile: